source/jquery.flot.drawSeries.js

```diff
@@ -39,22 +39,20 @@ This plugin is used by flot for drawing lines, plots, bars or area.
                 if(steps){
                     if (mx !== null && my !== null) {
                         // if middle point exists, transfer p2 -> p1 and p1 -> mp
-                        x2 = x1;
-                        y2 = y1;
                         x1 = mx;
                         y1 = my;
 
                         // 'remove' middle point
                         mx = null;
                         my = null;
-
-                        // subtract pointsize from i to have current point p1 handled again
-                        i -= ps;
                     } else if (y1 !== y2 && x1 !== x2) {
                         // create a middle point
                         y2 = y1;
                         mx = x2;
                         my = y1;
+
+                        // subtract pointsize from i to have current point p1 handled again
+                        i -= ps;
```

Comment:
Does [plotLineArea](https://github.com/flot/flot/blob/master/source/jquery.flot.drawSeries.js#L147) not need the same change?

Reply:
You might be right. However, I haven't been working on the project anymore for more than 2 months nor will I ever be, again, so unfortunately I have no easy way of testing it, or adding the tests.